okhttp-urlconnection/src/test/java/okhttp3/UrlConnectionCacheTest.java

```diff
@@ -1399,22 +1405,46 @@ private RecordedRequest assertClientSuppliedCondition(MockResponse seed, String
     assertEquals("B", readAscii(urlFactory.open(server.url("/").url())));
   }
 
+  public static String getPlatform() {
+    return System.getProperty("okhttp.platform", "platform");
+  }
+
+  @Test(expected = UnsupportedOperationException.class)
```

Comment:
I loathe expected exceptions as implemented in JUnit. They are clumsy for many reasons:
 - you can’t be confident that it threw on the last line of the test
 - you can’t assert the contents of the exception
 - you can’t assert the state of the world after the exception is thrown

Please just use `try { call(); fail(); } catch (UnsupportedOperationException expected) {}`